### PR TITLE
PIM-5071 : refactor spin method

### PR DESCRIPTION
**Pull request by jmleroux**
| Q                 | A
| ----------------- | ---
| Specs             |n
| Behats            |y
| Blue CI           |
| Changelog updated |n
| Review and 2 GTM  |y
| Micro Demo (PO)   |n
| Migration script  |n
| Tech Doc          |n

What's new : 
- timeout is only fixed in behat config file and accessed statically to avoid the redundant constant
- spinning is now "time based" and no more based on a number of iterations. Necessary because we are not really able to catch Timeout exceptions in nested spins

### CODE REVIEW

File: features/Context/FeatureContext.php [hunk 1]
```diff
@@ -432,4 +426,14 @@ public function getMailRecorder()
     {
         return $this->getContainer()->get('pim_enrich.mailer.mail_recorder');
     }
+
+    /**
+     * Set the waiting timeout
+     *
+     * @param $parameters
+     */
+    protected function setTimeout($parameters)
```
Comment (phaseinducer) [inline features/Context/FeatureContext.php]: Shouldn't it be static ?

Comment (jmleroux) [inline features/Context/FeatureContext.php]: It could be.

Comment (jmleroux) [inline features/Context/FeatureContext.php]: It could be

**Comment by sumbobyboys** [comment]
GTM

**Comment by phaseinducer** [comment]
GTM

**Comment by jjanvier** [comment]
GTM

File: features/Context/NavigationContext.php [hunk 2]
```diff
@@ -758,7 +758,7 @@ public function openPage($page, array $options = [])
             }
 
             return $page->verifyAfterLogin();
-        });
+        }, 'Trying to opening page ' . $this->currentPage);
```
Comment (jjanvier) [inline features/Context/NavigationContext.php]: open

**Pull request merged**